File: app/src/main/java/com/chydee/notekeeper/model/Note.java
package com.chydee.notekeeper.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "note_table")
public class Note {

    @PrimaryKey(autoGenerate = true)
    private int id;
    private String title;
    private String description;
    private int priority;

    //Constructor


    public Note(String title, String description, int priority) {
        this.title = title;
        this.description = description;
        this.priority = priority;
    }

    //Setter methods

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public int getPriority() {
        return priority;
    }
}

File: app/src/main/java/com/chydee/notekeeper/data/model/Note.kt
package com.chydee.notekeeper.data.model

import android.os.Parcelable
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.android.parcel.Parcelize

@Parcelize
@Entity(tableName = "note")
data class Note(
        @PrimaryKey(autoGenerate = true)
        @ColumnInfo(name = "id")
        val noteId: Int = 0,

        @ColumnInfo(name = "note_title")
        val noteTitle: String,

        @ColumnInfo(name = "note_detail")
        val noteDetail: String,

        @ColumnInfo(name = "last_edit")
        val lastEdit: String,

        @ColumnInfo(name = "is_note_locked")
        val isLocked: Boolean,

        @ColumnInfo(name = "color")
        val color: Int,

        @ColumnInfo(name = "password")
        val password: String?
) : Parcelable

